massa-cipher/src/encrypt.rs

```diff
@@ -10,14 +10,19 @@ use pbkdf2::password_hash::Salt;
 use pbkdf2::{password_hash::PasswordHasher, Pbkdf2};
 use rand::{distributions::Alphanumeric, thread_rng, Rng, RngCore};
 
-use crate::constants::{HASH_PARAMS, NONCE_SIZE, SALT_SIZE, VERSION};
+use crate::constants::{HASH_PARAMS, NONCE_SIZE, SALT_SIZE};
 use crate::error::CipherError;
-use massa_serialization::{Serializer, U32VarIntSerializer};
+
+pub struct CipherData {
+    pub salt: [u8; SALT_SIZE],
+    pub nonce: [u8; NONCE_SIZE],
+    pub encrypted_bytes: Vec<u8>,
+}
 
 /// Encryption function using AES-GCM cipher.
 ///
 /// Read `lib.rs` module documentation for more information.
-pub fn encrypt(password: &str, data: &[u8]) -> Result<Vec<u8>, CipherError> {
+pub fn encrypt(password: &str, data: &[u8]) -> Result<CipherData, CipherError> {
     // generate the PBKDF2 salt
     let raw_salt: String = thread_rng()
         .sample_iter(&Alphanumeric)
```

Comment:
does not follow spec: must be a sequence of bytes, not only alphanumeric which has way less entropy

Reply:
note that the lib expects us to base64-encode the bytes to generate the salt

Reply:
The problem is that the crate we use only accept &str and so utf8 chain of char which will not be the case if we use total random bytes

Reply:
I think that our crate expects us to provide `base64(random_bytes)`. Can you try and see if encryption on thyra + decryption in node works (and the opposite) works ?